chopper_generator/lib/src/generator.dart

```diff
@@ -466,10 +488,41 @@ final class ChopperGenerator
     });
   }
 
+  static void _cleanMapBody(List<Code> blocks, String refName,
+      {bool valueToString = false, DartType? type}) {
+    if (type != null) {
+      final mapTypeChecker = TypeChecker.fromRuntime(Map);
+      if (!mapTypeChecker.isExactlyType(type)) {
+        return;
+      }
+    }
+    final mapType = type?.toString() ?? 'Map<String, dynamic>';
+    final idNullableValueType =
+        mapType.endsWith('dynamic>') || mapType.endsWith('?>');
+    final isStringValueType = mapType.endsWith('String>');
+    if (!idNullableValueType && isStringValueType) {
+      return;
+    }
+    final cleanBody = '${refName}Clean';
+    blocks.add(Code('final $mapType $cleanBody = {};'));
+    blocks.add(Code('$refName.forEach((key, value) {'));
```

Comment:
`forEach` is not always preferred. See [lint rule](https://dart.dev/tools/linter-rules/avoid_function_literals_in_foreach_calls).